### PR TITLE
Changes output['params'] from tuple to dict, enhancing output express…

### PR DESCRIPTION
**Pull request by jeremy-myers**
…iveness and ability to pass output['params'] to cp_als() by unpacking operator **.

----
:books: Documentation preview :books:: https://pyttb--238.org.readthedocs.build/en/238/

### CODE REVIEW

File: tests/test_cp_als.py [hunk 1]
```diff
@@ -206,3 +206,23 @@ def test_cp_als_sptensor_zeros(capsys):
     capsys.readouterr()
     assert pytest.approx(output3["fit"], 1) == 0
     assert output3["normresidual"] == 0
+
+
+@pytest.mark.indevelopment
+def test_cp_als_tensor_pass_params(capsys, sample_tensor):
+    (data, T) = sample_tensor
```
Comment (ntjohnson1) [inline tests/test_cp_als.py:213]: Same tuple note from other PR

File: tests/test_cp_als.py [hunk 2]
```diff
@@ -206,3 +206,23 @@ def test_cp_als_sptensor_zeros(capsys):
     capsys.readouterr()
     assert pytest.approx(output3["fit"], 1) == 0
     assert output3["normresidual"] == 0
+
+
+@pytest.mark.indevelopment
```
Comment (ntjohnson1) [inline tests/test_cp_als.py:211]: Same mark note from other PR

File: tests/test_cp_als.py [hunk 3]
```diff
@@ -206,3 +206,23 @@ def test_cp_als_sptensor_zeros(capsys):
...
+def test_cp_als_tensor_pass_params(capsys, sample_tensor):
+    (data, T) = sample_tensor
+    KInit = ttb.ktensor.from_function(np.random.random_sample, T.shape, 2)
+
+    (M, Minit, output) = ttb.cp_als(T, 2, init=KInit)
+    capsys.readouterr()
+    assert pytest.approx(output["fit"]) == 1
```
Comment (ntjohnson1) [inline tests/test_cp_als.py:218]: Same note about not testing for fit and reducing iters

File: tests/test_cp_als.py [hunk 4]
```diff
@@ -206,3 +206,23 @@ def test_cp_als_sptensor_zeros(capsys):
...
+
+    # passing the same parameters back to the method will yield the exact same results
+    (M, Minit, output) = ttb.cp_als(T, 2, init=KInit, **output["params"])
```
Comment (ntjohnson1) [inline tests/test_cp_als.py:221]: Can we test here that the initial output dict matches the one returned for consistency?

**Review by ntjohnson1 — COMMENTED** [review]
Thanks. A similar few small requests and this also looks good to be merged by @dmdunla

**Review by ntjohnson1 — APPROVED** [review]

**Comment by dmdunla** [comment]
I do not understand the workflow here. This is merging a forked branch into a pyttb branch, but there is no pyttb PR to merge that branch:
- #236

**Comment by dmdunla** [comment]
@jeremy-myers and @ntjohnson1 WARNING: I'm going to change the target branch for this PR from [236](https://github.com/sandialabs/pyttb/tree/236-cp_als-does-not-accept-params-tuple-as-input) to [main](https://github.com/sandialabs/pyttb), but it says that commits may be dropped.

@jeremy-myers The target branch for PRs should be `main`, even when coming from a branch from your fork of pyttb. I will update the dev docs to help clarify this for future development.

**Comment by dmdunla** [comment]
@jeremy-myers, @ntjohnson1 FYI, here's the warning I see: `Some commits from the old base branch may be removed from the timeline, and old review comments may become outdated.`

**Comment by ntjohnson1** [comment]
I think it's a `may` rather than will. I think all these branches are relatively close to main so should be minimal risk. Will take a quick look. This one looks fine still.

**Comment by dmdunla** [comment]
@ntjohnson1 Except for the conflict? 😅

**Comment by ntjohnson1** [comment]
> @ntjohnson1 Except for the conflict? 😅

... oops. It should be fixed now. Hopefully that didn't spam too many emails.

**Pull request merged**